opencensus/stats/exporters/prometheus_exporter.py

```diff
@@ -183,26 +184,30 @@ def to_metric(self, desc, tag_values, agg_data):
                         aggregation_data_module.DistributionAggregationData):
 
             assert(agg_data.bounds == sorted(agg_data.bounds))
-            points = {}
-            cum_count = 0
+            # buckets are a list of bucket. Each bucket is another list with
+            # a pair of bucket name and value, or a triple of bucket name,
+            # value, and exemplar. buckets need to be in order.
+            buckets = []
+            cum_count = 0  # Prometheus buckets expect cumulative count.
             for ii, bound in enumerate(agg_data.bounds):
                 cum_count += agg_data.counts_per_bucket[ii]
-                points[str(bound)] = cum_count
+                bucket = [str(bound), cum_count]
```

Comment:
Using lists is kind of overkill and potentially expensive.
Why not tuples that properly match the grouping we are using of pairs, or triples?
```go
cur = (str(bound), cum_count,)
buckets = append(cur)
```

Reply:
Used list because Prometheus client comments said explicitly they expected a list of lists as buckets (https://github.com/prometheus/client_python/blob/master/prometheus_client/metrics_core.py#L201):
> buckets: A list of lists. Each inner list can be a pair of bucket name and value, or a triple of bucket name, value, and exemplar. The buckets must be sorted, and +Inf present.

Guess tuples will work in the same way. Updated.

Reply:
@songy23 oh I see, I didn't know that, but perhaps please cite that in your commit. My apologies, please revert to the original that you had as I was assuming that you had just used list of list out of convenience.

Reply:
No worries - changed it back to lists.